src/frame/mod.rs

```diff
@@ -231,7 +242,10 @@ impl From<RequestPdu> for Request {
 
 /// Represents a message from the server (slave) to the client (master).
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub(crate) struct ResponsePdu(pub(crate) Result<Response, ExceptionResponse>);
+pub struct ResponsePdu(pub(crate) Result<Response, ExceptionResponse>);
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct OptionalResponsePdu(pub(crate) Option<ResponsePdu>);
```

Comment:
Why is this type exposed unconditionally if it is only needed for the "server" feature?

Reply:
good point, I have to say I'm not yet completely accustomed to just how flexible the feature system is in rust and it didn't even occur me that it was possible to conditionally publish it, I'll fix it later

Reply:
hmmm, I actually ended up making the whole declaration only with the server feature :shrug: